### PR TITLE
plans: avoid next append overwriting last data

### PR DESCRIPTION
**Pull request by siddontang**
Fix issue #266

### CODE REVIEW

**Comment by siddontang** [comment]
@qiuyesuifeng

File: plan/plans/join.go [hunk 1]
```diff
@@ -213,11 +213,20 @@ func (r *JoinPlan) findMatchedRows(ctx context.Context, row *plan.Row, right boo
...
+		//   s1 = make([]interface{}, 0, 1)
+		//   s = append(s1, []interface{}{1})
+		//   ss = append(ss, s)
+		//   s = append(s1, []interface{}{2})
+		//   ss = append(ss, s)
+		// We will see that ss only contains 2.
+		joined := make([]interface{}, 0, len(cmpRow.Data)+len(row.Data))
 		if right {
-			joined = append(cmpRow.Data, row.Data...)
+			joined = append(append(joined, cmpRow.Data...), row.Data...)
```
Comment (qiuyesuifeng) [inline plan/plans/join.go]: Here can directly use copy?

Comment (siddontang) [inline plan/plans/join.go]: copy uses two lines, here just one line.

**Comment by qiuyesuifeng** [comment]
LGTM
Mybatis test passed.

**Comment by shenli** [comment]
LGTM

**Pull request merged**